public/javascripts/ood_shell.js

```diff
@@ -55,53 +8,52 @@ function OodShell(element, url, prefs) {
   this.term    = null;
 }
 
+/**
+ * Create the terminal. The constructor for the OodShell class is mostly just a data object. This function
+ * does the actual work of initializing the hterm.Terminal, the websocket (through a callback *after* the 
+ * terminal is ready.) and setting preferences, etc.
+ */
 OodShell.prototype.createTerminal = function () {
-  if(lib.resource.get('hterm/changelog/version', null).data === '1.80') {
-    this.socket = new WebSocket(this.url);
-    this.socket.onopen    = this.runTerminal.bind(this);
-    this.socket.onmessage = this.getMessage.bind(this);
-    this.socket.onclose   = this.closeTerminal.bind(this);
-  } else {
-    document.getElementById('terminal').innerHTML = 'WARNING: version of hterm has changed. Please review ood_shell.js to ensure that the override of hterm.Terminal.prototype.copyStringToClipboard is still an appropriate fix to copy on select in Firefox.';
-  }
-};
-
-OodShell.prototype.runTerminal = function () {
-  var that = this;
-
   // Set backing store that hterm uses to read/write preferences
   hterm.defaultStorage = new lib.Storage.Memory();
-
+  
   // Create an instance of hterm.Terminal
   this.term = new hterm.Terminal();
-
-  // Set preferences for terminal
-  for (var k in this.prefs) {
-    this.term.prefs_.set(k, this.prefs[k]);
-  }
-
+    
+  var that = this;
+    
   // Handler that fires when terminal is initialized and ready for use
   this.term.onTerminalReady = function () {
     // Create a new terminal IO object and give it the foreground.
     // (The default IO object just prints warning messages about unhandled
     // things to the JS console.)
     var io = this.io.push();
-
+  
```

Comment:
Careful with your newlines. This added spaces that clutter the diff.

Reply:
Yea it's weird diff issue. If you see the raw file there's only the 1 empty line. I think that git just got confused bc left is line 89 and right is line 31.

Reply:
The raw file has one empty line but the issue here is that the 2 spaces have been added:

https://github.com/OSC/ood-shell/blob/ecd82a5b45715fcf5a6d54b38d4a76b07a9e1566/public/javascripts/ood_shell.js#L19

Reply:
I have vim configured to auto delete trailing whitespace of lines, so that is one approach to avoid these in the future.

Reply:
Right, ok, I see.  I made all these edits in our file editor. I'll try clean all these up now.